mono/mini/main.c

```diff
@@ -179,8 +182,49 @@ probe_embedded (const char *program, int *ref_argc, char **ref_argv [])
 		goto doclose;
 	if (read (fd, sigbuffer, sizeof (sigbuffer)) == -1)
 		goto doclose;
-	if (memcmp (sigbuffer+sizeof(uint64_t), "xmonkeysloveplay", 16) != 0)
+	if (memcmp (sigbuffer+sizeof(uint64_t), "xmonkeysloveplay", 16) == 0)
+		goto found;
+	else
```

Comment:
There's no need to put all this in an else block, just pull it out.

Reply:
Yes, there is - there is another `goto found` in the new code block. Basically the old code assume the xmonkeyloveplay is located at the end. The new code tries this first (the above block), and if not found, try harder by looking at the mac header and find it via the header from the tail end of the signature section.

Reply:
So the "found" section is now jumped to, from two places, instead of one.

Reply:
https://gist.github.com/CoffeeFlux/6e68c33b9dfae5ce305ce358ec1d44d5 Why does this not work? Am I missing something?

Reply:
Okay, I see. I guess it is a matter of style/preference: I don't like bare / unscoped `goto`'s. If I were to modify the code to your direction, I'd put a comment next to the bare `goto` something like `// did not find "..." at either of two places`, as a reminder.

The header scanning code is quite mac os x specific and have a few "local" variables, like the header structs, not used outside, so it is somewhat nice to put the code block in a {}, even with #defines to exclude it for other OSes.

But the two probably compile to identical codes.

Reply:
If you feel strongly about it that's fine - not my preference but it should ultimately product the same output.